### PR TITLE
flutter update-packages --force-upgrade + analyzer fix

### LINKED ISSUE #108184: Updating to `analyzer 4.3.0` shows new `super-parameters` errors
Updating to `analyzer 4.3.0` causes `super-parameters` errors:

https://logs.chromium.org/logs/flutter/buildbucket/cr-buildbucket/8807915499059549089/+/u/Analyze/analyze/test_stdout
```
skipping /b/s/w/ir/x/w/flutter sdk/packages/flutter/lib/src/material/expansion_tile_theme.dart: Invalid argument(s): Content produced diagnostics when parsed:
  EXPERIMENT_NOT_ENABLED: This requires the 'super-parameters' language feature to be enabled. - 186:5
  EXPERIMENT_NOT_ENABLED: This requires the 'super-parameters' language feature to be enabled. - 188:14

skipping /b/s/w/ir/x/w/flutter sdk/packages/flutter/lib/src/material/switch_theme.dart: Invalid argument(s): Content produced diagnostics when parsed:
  EXPERIMENT_NOT_ENABLED: This requires the 'super-parameters' language feature to be enabled. - 165:5
  EXPERIMENT_NOT_ENABLED: This requires the 'super-parameters' language feature to be enabled. - 167:14

skipping /b/s/w/ir/x/w/flutter sdk/packages/flutter/lib/src/material/ink_well.dart: Invalid argument(s): Content produced diagnostics when parsed:
  EXPERIMENT_NOT_ENABLED: This requires the 'super-parameters' language feature to be enabled. - 38:14
  EXPERIMENT_NOT_ENABLED: This requires the 'super-parameters' language feature to be enabled. - 39:14
```
etc

Upgraded in https://github.com/flutter/flutter/pull/108178/

### PR DESCRIPTION
**Pull request by goderbauer**
Supersedes https://github.com/flutter/flutter/pull/108178.
Fixes https://github.com/flutter/flutter/issues/108184.

First commit is result of `flutter update-packages --force-upgrade`. 
Second commit is the fix for the analyzer errors (https://github.com/flutter/flutter/issues/108184).
Third commit fixes newly surfaced `avoid_redundant_argument_values` warnings.

### CODE REVIEW

**Comment by goderbauer** [comment]
Looks like the analyzer found some new different warnings now. I'll fix those up in a sec.

**Comment by goderbauer** [comment]
The analyzer check is happy now. 🎊

File: dev/bots/analyze.dart [hunk 1]
```diff
@@ -204,9 +203,7 @@ Future<void> run(List<String> arguments) async {
 
 // TESTS
 
-FeatureSet _parsingFeatureSet() => FeatureSet.fromEnableFlags2(
-    sdkLanguageVersion: Version.parse('2.17.0-0'),
-    flags: <String>['super-parameters']);
+FeatureSet _parsingFeatureSet() => FeatureSet.latestLanguageVersion();
```
Comment (jmagman) [inline dev/bots/analyze.dart:206]: 🎉

**Review by jmagman — APPROVED** [review]
LGTM!

**Comment by jmagman** [comment]
Need to rebase onto #107984 which rolled `archive`.

**Pull request merged**